File: lightbulb-easyrecyclerview/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerview/decorations/FlexboxSpaceItemDecoration.java
package com.github.rooneyandshadows.lightbulb.easyrecyclerview.decorations;

import android.graphics.Rect;
import android.view.View;

import com.github.rooneyandshadows.lightbulb.easyrecyclerview.EasyRecyclerView;
import com.github.rooneyandshadows.lightbulb.easyrecyclerview.R;
import com.google.android.flexbox.FlexboxLayoutManager;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

@SuppressWarnings("rawtypes")
public class FlexboxSpaceItemDecoration extends RecyclerView.ItemDecoration {
    private final int verticalSpacing;
    private final int horizontalSpacing;

    public FlexboxSpaceItemDecoration(int spacing, EasyRecyclerView recyclerView) {
        this.verticalSpacing = spacing / 2;
        this.horizontalSpacing = spacing / 2;
        recyclerView.findViewById(R.id.recyclerView).setPadding(horizontalSpacing, spacing, horizontalSpacing, verticalSpacing);
    }

    public FlexboxSpaceItemDecoration(int verticalSpacing, int horizontalSpacing, EasyRecyclerView recyclerView) {
        this.verticalSpacing = verticalSpacing / 2;
        this.horizontalSpacing = horizontalSpacing / 2;
        recyclerView.findViewById(R.id.recyclerView).setPadding(this.verticalSpacing, verticalSpacing, this.verticalSpacing, this.horizontalSpacing);
    }

    @Override
    public void getItemOffsets(Rect outRect, @NonNull View view, RecyclerView parent, @NonNull RecyclerView.State state) {
        outRect.left = horizontalSpacing;
        outRect.right = horizontalSpacing;
        outRect.bottom = verticalSpacing * 2;
    }
}

File: lightbulb-easyrecyclerview/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerview/item_decorations/FlexboxSpaceItemDecoration.kt
package com.github.rooneyandshadows.lightbulb.easyrecyclerview.item_decorations

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.github.rooneyandshadows.lightbulb.easyrecyclerview.EasyRecyclerView
import com.github.rooneyandshadows.lightbulb.easyrecyclerview.R
import com.github.rooneyandshadows.lightbulb.easyrecyclerview.item_decorations.base.EasyRecyclerItemDecoration

@Suppress("unused")
class FlexboxSpaceItemDecoration : EasyRecyclerItemDecoration {
    private val verticalSpacing: Int
    private val horizontalSpacing: Int

    constructor(spacing: Int, recyclerView: EasyRecyclerView<*>) {
        verticalSpacing = spacing / 2
        horizontalSpacing = spacing / 2
        recyclerView.findViewById<View>(R.id.recyclerView).setPadding(horizontalSpacing, spacing, horizontalSpacing, 0)
    }

    constructor(verticalSpacing: Int, horizontalSpacing: Int, recyclerView: EasyRecyclerView<*>) {
        this.verticalSpacing = verticalSpacing / 2
        this.horizontalSpacing = horizontalSpacing / 2
        recyclerView.findViewById<View>(R.id.recyclerView)
            .setPadding(this.verticalSpacing, verticalSpacing, this.verticalSpacing, 0)
    }

    @Override
    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        outRect.left = horizontalSpacing
        outRect.right = horizontalSpacing
        outRect.bottom = verticalSpacing * 2
    }
}